src/libs/actions/Modal.ts

```diff
@@ -1,30 +1,38 @@
 import Onyx from 'react-native-onyx';
 import ONYXKEYS from '@src/ONYXKEYS';
 
-let closeModal: (isNavigating: boolean) => void;
+const closeModals: Record<number, null | ((isNavigating: boolean) => void)> = {};
+let count = 0;
+
+// let closeModal: (isNavigating: boolean) => void;
 let onModalClose: null | (() => void);
 
+/**
+ * Get the available key that we can store the onClose callback into it
+ */
+function getAvailableKey() {
```

Comment:
This function increases a number to make new keys. This could cause problems if we make a lot of keys over time or if many things happen at once.